feat: 계좌 해지 시 잔액 송금 및 복구 기능 구현

try3982 opened the pull request:
## #️⃣ Issue Number
- #24 
<!--- ex) #이슈번호, #이슈번호 -->

## 📝 요약(Summary)

<!--- 변경 사항 및 관련 이슈에 대해 간단하게 작성해주세요. 어떻게보다 무엇을 왜 수정했는지 설명해주세요. -->

## 🛠️ PR 유형

어떤 변경 사항이 있나요?

- [x] 새로운 기능 추가

## 개발 상세 내역

### ✅  1.계좌 해지 시 잔액 자동 송금 처리
- deleteAccount(Long userId, String closingAccountNumber, String withdrawAccountNumber) 메서드 구현
- 해지할 계좌의 잔액을 출금 계좌로 이체
- 해지한 계좌 상태를 CLOSED로 변경 후 저장
- 출금 계좌와 사용자 소유 일치 및 상태 검증 로직 추가

### ✅  2.계좌 복구 기능 구현
- /account/restore API 구현
- 3개월 이내 복구 시 계좌 상태를 ACTIVE로 변경
- 3개월 초과 시 해당 계좌 영구 삭제 처리
- Account에 unregisteredAt 필드 추가 및 해지 시 저장

### ✅  3.계좌 개설 시 최소 예치금 검증 강화
- validateInitialBalance() 로직 수정
   - initialBalance < accountType.getBalance() 조건으로 검증

### ✅  4.ENUM 추가
- WITHDRAW_ACCOUNT_UNMATCH("출금 계좌가 본인 계좌가 아닙니다.")
- WITHDRAW_ACCOUNT_INACTIVE("출금 계좌가 사용 불가능한 상태입니다.")
- ACCOUNT_NOT_RECOVERABLE("계좌 복구 가능 기간이 지났습니다.")

try3982 committed:
feat: 계좌 해지 시 잔액 송금 및 복구 기능 구현

- 계좌 해지 시 잔액을 등록된 출금 계좌로 자동 송금 처리
- 계좌 상태를 CLOSED로 변경하고 해지 일시(unregisteredAt) 저장
- 해지된 계좌는 3개월 이내 복구 가능, 초과 시 영구 삭제 처리
- 계좌 복구 API(/account/restore) 추가
- 최소 예치금 미만일 경우 계좌 개설 불가 로직 수정

Paransaik commented on src/main/java/com/track/fin/controller/AccountController.java:
```diff
@@ -61,4 +62,14 @@ public Account getAccount(
         return accountService.getAccount(id);
     }
+
+    @PostMapping("/account/restore")
```
이전 설명드린 API 명명 규칙에 따라 수정하는게 좋을 거 같습니다.

`@PostMapping("/accounts/{account_id}/restore")`

Paransaik commented on src/main/java/com/track/fin/domain/Account.java:
```diff
@@ -73,4 +75,14 @@ public void withdraw(Long amount) {
     }
 
+    public void close() {
+        this.accountStatus = AccountStatus.CLOSED;
```
지난 번 Enum 사용 시 static import 방식을 사용하기로 정했습니다.

Paransaik commented on src/main/java/com/track/fin/service/AccountService.java:
```diff
@@ -75,19 +77,23 @@ public List<Account> getAccounts(Long userId) {
+            transactionService.transfer(userId, accountNumber, withdrawAccountNumber, closingAccount.getBalance());
+        }
+
+        closingAccount.setAccountStatus(CLOSED);
```
위에 close 라는 함수를 만드셨는데, 왜 AccountStatus를 set으로 하시는지 알 수 있을까요?

Paransaik commented:
1. 송금 완료 후 계좌 상태를 'FROZNE'으로 변경 
    - 해당 이슈는 어떤식으로 구현하셨나요?

2. 복구 기한 초과 시 계좌는 영구 삭제
    - 해당 이슈는 어떻게 구현하실 예정이신가요?

3. PR 유형에 체크박스 오타 있습니다.

try3982 commented:
1. 오타입니다. 송금완료후 CLOSED로 변경되게 구현하였습니다.
2. restoreAccount() 메서드 내부에 복구 기한이 지난 계좌는 DB에서 삭제하도록 구현하였습니다.
3. 오타 수정하였습니다.

try3982 committed:
refactor: 리뷰반영

-. User도메인 loginId 변수명 변경
-. restoreAccount url 변경
-. emum import 방식으로 사용하도록 변경

try3982 committed:
refactor: 사용하지 않는 import문 삭제

try3982 committed:
Merge branch 'dev' into feat/create-and-close-account

try3982 committed:
refactor: 개행

try3982 committed:
refactor:

try3982 committed:
Merge remote-tracking branch 'origin/feat/create-and-close-account' into feat/create-and-close-account

Paransaik commented on src/main/java/com/track/fin/domain/Account.java:
```diff
@@ -71,13 +78,25 @@ public void withdraw(Long amount) {
     }
 
     public void restore() {
         this.accountStatus = AccountStatus.ACTIVE;
```
위와 같이 static import enum을 사용하기 때문에 AccountStatus 빼주셔야 합니다.

Paransaik commented on src/main/java/com/track/fin/service/AccountService.java:
```diff
@@ -84,10 +84,12 @@ public AccountDto deleteAccount(Long userId, String accountNumber, String withdr
         validateDeleteAccount(user, closingAccount, withdrawAccount);
 
+        // 여기서 검증할 필요가 있나?
         if (closingAccount.getBalance() > 0) {
```
지난번 회의 때 의견 나누었으니 transactionService에게 조건 검사에 대한 역할을 위힘하였으므로 분기 처리는 빼기로 합시다.

Paransaik commented on src/main/java/com/track/fin/service/AccountService.java:
```diff
@@ -84,10 +84,12 @@ public AccountDto deleteAccount(Long userId, String accountNumber, String withdr
         }
 
+        closingAccount.close();
         closingAccount.setAccountStatus(CLOSED);
```
closiongAccount.close()안에서 set을 하시는데 아래는 지워야하지 않을까요?...

Paransaik reviewed (commented):
초반에 비해 코드 컨벤션과 가독성이 향상되었으나 같은 이유로 PR이 달립니다.

Commit 전 조금 더 신경써주시면 좋을 거 같습니다.

try3982 committed:
refactor: 리뷰 반영
-. close 삭제
-. enum import로 사용
-.담보금액 이외 사용가능하도록 구현
